lib/Arch/Sleigh/Arch.cpp

```diff
@@ -519,9 +367,94 @@ std::optional<int32_t> SingleInstructionSleighContext::oneInstruction(
 }
 
 
-OperandLifter::OpLifterPtr
-SleighArch::DefaultLifter(const remill::IntrinsicTable &intrinsics) const {
-  return std::make_unique<SleighLifter>(this, intrinsics);
+namespace {
+
+template <typename... Ts>  // (7)
+struct Overload : Ts... {
+  using Ts::operator()...;
+};
+template <class... Ts>
+Overload(Ts...) -> Overload<Ts...>;
+
+}  // namespace
+
+void SleighDecoder::ApplyFlowToInstruction(remill::Instruction &inst) const {
```

Comment:
Maybe make this of type `Category ConvertFlow( Flow, next_pc )` ? That was it is a pure function in a sense, which is usually easier to compose/reason about. Also it makes explicit what parts of `Instruction` it cares about.

Reply:
So we also have to setup the branch_taken not_taken pc for the flows for which this is applicable so. im not sure this type makes sense